x/delayedack/ibc_middleware.go

```diff
@@ -46,10 +46,11 @@ func (im IBCMiddleware) OnRecvPacket(
 	if !im.keeper.IsRollappsEnabled(ctx) {
```

Comment:
Just a thought: `OnRecvPacket`, `OnAcknowledgementPacket` and `OnTimeoutPacket` have a lot in common, it's a shame the code is not reused 😄

Reply:
true. but now we're in the business of solving bugs not optimzing :) I will create an issue for it to optimize.

Reply:
Of course, we could easily shoot ourselves in the foot with the refactor

Reply:
![image](https://github.com/dymensionxyz/dymension/assets/20242145/069ea65e-e598-4cdf-9219-c26c3be15776)